wrapper

```diff
@@ -106,60 +107,45 @@ def download_file(ticket, src):
 
 # Download a set of files referenced in a ticket list file from iRODS, returning a list of downloaded files.
 def download_files(ticket_list_path):
+    downloaded_files = [];
     with TicketListReader(ticket_list_path) as tlr:
         for ticket, src in tlr:
             download_file(ticket, src)
+            downloaded_files += [os.path.basname(src)]
+    return downloaded_files
 
-# Run the gene length job.
+# Run the job.
 def run_job(arguments, output_filename, error_filename):
-    output_files = []
     with open(output_filename, "w") as out, open(error_filename, "w") as err:
-        for input_file in arguments:
-            output_file = re.sub(r"[.][^.]+$", ".txt", input_file)
-            try:
-                with open(input_file, "rU") as fh_in, open(output_file, "w") as fh_out:
-                    get_gene_lengths(fh_in, fh_out)
-                    output_files += [output_file]
-            except Exception as e:
-                msg = "error processing {0}: {1}".format(input_file, e)
-                eprint(msg)
-                raise Exception(msg)
-    return output_files
-
-# Upload a file or directory to iRODS.
-def upload_file(ticket, irods_user, owner, src, dest):
-    rc = subprocess.call(["iput", "-rt", ticket, src, dest])
-    if rc != 0:
-        raise Exception("could not upload {0} to {1}".format(src, dest))
-
-    # Don't bother modifying the file permissions if the iRODS user and file owner are the same.
-    if irods_user == owner:
-        return
-
-    # Update the file permissions so that the person who submitted the job can see the output files.
-    fullpath = os.path.join(dest, src)
-    rc = subprocess.call(["ichmod", "own", owner, fullpath])
-    if rc != 0:
-        raise Exception("could not change the owner of {0} to {1}".format(fullpath, owner))
-    rc = subprocess.call(["ichmod", "null", irods_user, fullpath])
-    if rc != 0:
-        raise Exception("could not remove {0} permissions on {1}".format(irods_user, fullpath))
+        rc = subprocess.call(["Hisat2-Cuffcompare-Cuffmerge.sh"] + arguments, stdout=out, stderr=err)
+        if rc != 0:
+            raise Exception("Hisat2-Cuffcompare-Cuffmerge.sh returned exit code {0}".format(rc))
```

Comment:
`run_job` can be much simpler for this version of the script. `upload_file` has been replaced by the separate script.